tests/phpunit/integration/Core/Modules/ModulesTest.php

```diff
@@ -62,6 +62,44 @@ function ( $instance ) {
 		);
 	}
 
+	public function test_get_available_modules__missing_dependency() {
+		$modules = new Modules( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+
+		add_filter(
+			'googlesitekit_available_modules',
+			function( $modules ) {
+				return array_filter(
+					$modules,
+					function( $module ) {
+						// Remove Analytics from the list of available modules.
+						return 'analytics' !== $module;
+					}
+				);
```

Comment:
Let's provide a specific list here instead of using the filter otherwise this test will break when new modules are introduced.

Reply:
I am of the opinion that we should keep this as-is. My view is this test is somewhat complimentary to `test_get_available_modules`. Please take a look at `test_get_available_modules` and you'll notice how the assertion is similar, and when we add new modules, we will need to update `test_get_available_modules` - at which point we'd update `test_get_available_modules__missing_dependency` too.

I think this would be a reasonable approach, but please take a look and see what you think.